app/services/products/fetch.rb

```diff
@@ -0,0 +1,13 @@
+module Shop
+  module Services
+    module Products
+      class Fetch
+        def call(product_id:)
+          product = PRODUCTS.find { |product| product.id == product_id }
```

Comment:
```
app/services/products/fetch.rb:6:38: W: Shadowing outer local variable - product.
          product = PRODUCTS.find { |product| product.id == product_id }
                                     ^^^^^^^
```

You used the same variable name for local variable and variable inside block - it isn't good